lldb/test/API/lang/swift/async/unwind/unwind_in_all_instructions/TestSwiftAsyncUnwindAllInstructions.py

```diff
@@ -0,0 +1,134 @@
+            )
+            function = sym_ctx_list[0].function
+
+            instructions = list(function.GetInstructions(target))
+            self.assertGreater(len(instructions), 0)
+            for instruction in instructions:
+                bp = target.BreakpointCreateBySBAddress(instruction.GetAddress())
+                self.assertTrue(
+                    bp.IsValid(), f"failed to set bp inside funclet {funclet_name}"
+                )
+                breakpoints.add(bp.GetID())
+        return breakpoints
+
+    # FIXME: there are challenges when unwinding Q funclets ("await resume"),
+    # see rdar://137048317. For now, we only know how to unwind during and
+    # shortly after the prologue. This function returns "should skip" if we're
+    # at a PC that is too far from the prologue (~16 bytes). This is a
+    # rough approximation that seems to work for both x86 and arm.
+    def should_skip_Q_funclet(self, thread):
+        current_frame = thread.frames[0]
+        function = current_frame.GetFunction()
+        if "await resume" not in function.GetName():
+            return False
+
+        max_prologue_offset = 16
+        prologue_end = function.GetStartAddress()
+        prologue_end.OffsetAddress(function.GetPrologueByteSize() + max_prologue_offset)
+        current_pc = current_frame.GetPCAddress()
+        return current_pc.GetFileAddress() >= prologue_end.GetFileAddress()
+
+    def check_unwind_ok(self, thread, bpid):
+        if self.should_skip_Q_funclet(thread):
+            return
+        # Check that we see the virtual backtrace:
+        expected_funcnames = [
+            "ASYNC___1___",
+            "ASYNC___2___",
+            "ASYNC___3___",
+            "ASYNC___4___",
+            "ASYNC___5___",
+        ]
+        frames = thread.frames
+        self.assertGreater(
+            len(frames), len(expected_funcnames), f"Invalid backtrace for {frames}"
+        )
+        actual_funcnames = [
+            frame.GetFunctionName() for frame in frames[: len(expected_funcnames)]
+        ]
+        for expected_name, actual_name in zip(expected_funcnames, actual_funcnames):
+            self.assertIn(expected_name, actual_name, f"Unexpected backtrace: {frames}")
+
+    @swiftTest
+    @skipIf(oslist=["windows", "linux"])
+    def test(self):
+        """Test that the debugger can unwind at all instructions of all funclets"""
+        self.build()
+
+        source_file = lldb.SBFileSpec("main.swift")
+        target, process, _, _ = lldbutil.run_to_source_breakpoint(
+            self, "BREAK HERE", source_file
+        )
+
+        breakpoints = self.set_breakpoints_all_funclets(target)
+        num_breakpoints = len(breakpoints)
+
+        # Reach most breakpoints and ensure we can unwind in that position.
+        while True:
+            process.Continue()
+            thread = lldbutil.get_stopped_thread(process, lldb.eStopReasonBreakpoint)
+            if thread is None:
+                break
```

Comment:
when does this happen?

Reply:
If the process exits instead of breaking, for example.

Reply:
should that be the condition then?

Reply:
Let me try rewriting this